ext/SpeedyWeatherGeoMakieExt.jl

```diff
@@ -121,4 +122,180 @@ function SpeedyWeather.globe(
+
+# Example
+```julia
+using SpeedyWeather, GeoMakie, CairoMakie
+
+# Animate temperature at level 1
+animate_simulation("output.nc", variable="temp", level=1)
+
+# Animate surface pressure with a specific colormap
+animate_simulation("output.nc", variable="pres", colormap=:thermal)
+
+# Create an animation with Orthographic projection
+animate_simulation("output.nc", variable="temp", projection="+proj=ortho +lon_0=0 +lat_0=30")
+```
+"""
+function SpeedyWeather.animate_simulation(
+    nc_file::String;
+    variable::String = "temp",
+    level::Int = 1,
+    transient_timesteps::Int = 0,
+    output_file::String = "animation.mp4",
+    colormap = :viridis,
+    framerate::Int = 15,
+    title::String = "",
+    colorrange = nothing,
+    projection = "+proj=robin",
+    figure_size = (800, 600)
+)
+  
+    # Open the NetCDF file
+    ds = NCDataset(nc_file)
+    
+    # Get dimensions
+    lon = ds["lon"][:]
+    lat = ds["lat"][:]
+    time = ds["time"][:]
+    
+    # Get time units for proper labeling
+    time_units = ds["time"].attrib["units"]
+    
+    # Get variable metadata
+    var_long_name = get(ds[variable].attrib, "long_name", variable)
+    var_units = get(ds[variable].attrib, "units", "")
+    
+    # Set title if not provided
+    if isempty(title)
+        title = var_long_name
+    end
+    
+    # Check if the variable is 3D (has a vertical dimension)
+    is_3d = "layer" in dimnames(ds[variable])
+    
+    # Create the figure
+    fig = Figure(size = figure_size)
+    
+    # Create the axis
+    ax = GeoAxis(
+            fig[1, 1],
+            title = title,
+            titlesize = 20,
+            dest = projection
+        )
+    
+    # Create a placeholder for the plot
+    if is_3d
+        # For 3D variables, extract the specified level
+        data_init = ds[variable][:, :, level, transient_timesteps + 1]
+    else
+        # For 2D variables
+        data_init = ds[variable][:, :, transient_timesteps + 1]
+    end
+    
+    # Determine color range if not specified
+    if isnothing(colorrange)
+        # Sample data to determine a good color range
+        sample_indices = min(10, length(time))
+        
+        if is_3d
+            sample_data = ds[variable][:, :, level, 1:sample_indices]
+        else
+            sample_data = ds[variable][:, :, 1:sample_indices]
+        end
+        
+        # Calculate min and max across samples
+        sample_data = filter(!isnan, sample_data)
+        
+        if !isempty(sample_data)
+            data_min, data_max = extrema(sample_data)
+            # Add a small buffer to the range
+            range_buffer = (data_max - data_min) * 0.05
+            colorrange = (data_min - range_buffer, data_max + range_buffer)
+        else
+            colorrange = (-1, 1)  # Fallback if all data is NaN
+        end
+    end
+    
+    # Create the surface plot
+    hm = surface!(ax, lon, lat, data_init; colormap=colormap, colorrange=colorrange)
```

Comment:
I think there's a better function here than `surface!` which is meant for 3D plots and therefore has some additional lines that dim the colours of the plot through transparency. Maybe check `meshimage`?

Reply:
GeoMakie docs mostly use `surface!` for these kind of plots as well. I think `heatmap` and `meshimage` would be alternatives. But I don't have a strong opinion on that. Maybe just make it an option?